pytorch_lightning/loops/base.py

```diff
@@ -238,3 +240,70 @@ def _load_from_state_dict(
 
         self.on_load_checkpoint(state_dict[prefix + "state_dict"])
         self.restarting = True
+
+
+class ExternalLoop(Loop):
```

Comment:
as written above, i really think the loops should only deal with stuff within the trainer calls not external. everything that happens outside of the trainer class should be handled by utility classes (if at all) or left to the user/flash.

Reply:
I agree with @justusschock . There's also no bound on what options we provide here, and going in this direction will end up exposing all of the trainer internals to users, so i'm not in favor of this approach

Reply:
In the case of this kfold loop, do we need this class at all? the extra properties it defines could very well just live in the kfold loop itself as they are specific to the need there.